core/server/master/src/main/java/alluxio/master/file/S3AUploadCleaner.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file;
+
+import alluxio.AlluxioURI;
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.heartbeat.HeartbeatExecutor;
+import alluxio.util.UnderFileSystemUtils;
+import alluxio.wire.MountPointInfo;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.Protocol;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * S3A intermediate multipart upload periodic clean.
+ */
+@NotThreadSafe
+final class S3AUploadCleaner implements HeartbeatExecutor {
```

Comment:
Would it be cleaner (no pun intended) to put this logic in the S3 under filesystem?

Reply:
In the previous meeting, we thought it may be too many cleanup if we put it in the S3 under filesystem. But it would be definitely cleaner to put this logic in the S3AUnderFileSystem, we do not need to create additional transfer managers and could reuse nearly all of the codes.

Reply:
Put the cleanup logic in the `S3AUnderFileSystem.cleanup`. Leader master will clean up all UFS periodically if `UNDERFS_CLEANUP_ENABLED`. This makes the code much cleaner! Thanks